### DISCUSSION TITLE
Provide possibility to specify imprint, data privacy policy, terms of service, etc. for self-hosted installations

### DISCUSSION THREAD

**Discussion by b2cc** [lukevella/rallly#1387] [Ideas]
Thanks for providing this app, it's a really great alternative to similar cloud solutions!

For the self-hosted version it would be great if we had the option to specify links (or be able to provide the text directly in the application) for the terms of service, privacy policy etc. This is legally mandatory in many countries and would be really helpful if this was configurable.

Thanks for your consideration!

**Comment by lukevella** [comment]
Would be interested to see how this is implemented in other self-hosted apps as a reference.

**Reply by b2cc** [reply]
The way I know it from some of the Fediverse apps, usually they provide text field that allows to fill in all the relevant info. Often they support to render it either as HTML or Markdown to support some basic styling options.

See here for example:
* Lemmy: e.g. https://feddit.org/ - on the right hand side, middle (german) or at the bottom (english) there are links to the TOMs etc. This is based on some markdown fields which  can freely be specified in the app settings.
* Mastodon:  e.g. https://wien.rocks/about (also markdown)

But an option to specify some links to some other website where the pages can be hosted would also be completely acceptable I guess, and probably less work.

**Reply by SF2311** [reply]
I would love to have this feature as well!

In my opinion, the ability to provide custom links that point to a privacy statement/imprint hosted somewhere else is absolutely sufficient.
I see two options to achieve this:
* Either config options/environment variables for the two links (privacy policy and imprint) which are rendered in the footer if provided
* Two links in the footer, that are always rendered and point to "dummy" links on the configured domain (i.e., `poll.example.org/privacy`). 
   The rest could then be solved by a redirect using a reverse proxy.

I have also seen apps provide the possibility to add custom elements to the footer (See [Homer](https://github.com/bastienwirtz/homer/blob/main/docs/configuration.md) or [Engelsystem](https://github.com/engelsystem/engelsystem/blob/main/config/config.default.php#L45) for an example, but that would probably be a lot of effort to realize.
